src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -555,6 +568,15 @@ def get(self, args):
         # Check the command index for (command: [module]) mapping, like
         # "network": ["azure.cli.command_modules.natgateway", "azure.cli.command_modules.network", "azext_firewall"]
         index_modules_extensions = index.get(top_command)
+        if not index_modules_extensions and is_autocomplete():
+            # If user type `az acco`, command begin with `acco` will be matched.
+            # If the user just installed a new extension which contains a command named `acco`, he can't get the
+            # expected suggestion. But it's a rare scenario, and it's okay to ignore.
+            logger.debug("In autocomplete mode, load command begin with: '%s'", top_command)
+            index_modules_extensions = []
+            for command in index:
+                if command.startswith(top_command):
+                    index_modules_extensions += index[command]
```

Comment:
Maybe put these lines after L597 for simplicity?

Reply:
I put it here as the following code convert `index_modules_extensions` to `index_builtin_modules` and `index_extensions`